### ISSUE TITLE
SoftBody wind has unclear explanation

### ISSUE THREAD

**Issue by yankscally** [godotengine/godot-docs#9588]
**Your Godot version:**

4.1 beta2

**Issue description:**

I can blow the wind correctly, but I'm not sure how to make it blow in a set direction.

**URL to the documentation page:**

Softbody/Area3D wind 

If you know how to fix the issue you are reporting please
consider opening a pull request. We provide a tutorial on
using git here: https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html,
writing documentation at https://docs.godotengine.org/en/stable/community/contributing/docs_writing_guidelines.html
and contributing to the class reference here: https://docs.godotengine.org/en/stable/community/contributing/updating_the_class_reference.html

**Comment by yankscally** [comment]
If I can figure it out I can provide a better example of how to use this feature.

**Comment by AThousandShips** [comment]
The description is:
> The Node3D which is used to specify the direction and origin of an area-specific wind force. The direction is opposite to the z-axis of the Node3D's local transform, and its origin is the origin of the Node3D's local transform.

What do you feel is missing from this? Not sure how an example would help as it's just a node path to set up

**Comment by yankscally** [comment]
well, it could be broken, I'm not sure..

a better name would be "wind target", and the description could say, "wind force is negative-z to the wind target"

if not - the description doesn't really make sense, as I'm not able to get the effect described by using the target node. It just blows in one direction no matter what I'm doing.

**Comment by AThousandShips** [comment]
So it *doesn't* blow based on the axis? So if the local transform has the z-axis point to the right for example?

The nose isn't the target though, it's the source, and the direction is based on how it points, not where it is

To me the description is quite clear and literal, but there's always room for improvement

**Comment by yankscally** [comment]
OK, thank you for testing that. Great that its not a bug in the end. 

So the transform of the wind target node is not important, only the rotation. And it resets on_entered.. not sure what you meant by that. but we should include these things in the docs

**Comment by AThousandShips** [comment]
The rotation of the source node what matters, which is part of the transform, the translation doesn't control the direction though, specifically the direction of the z-axis, if you need more info on 3D transforms please read [this](https://docs.godotengine.org/en/latest/tutorials/3d/using_transforms.html)

To me the documentation says exactly what it does, and I don't know how to explain it any clearer, it might be that you misunderstood how 3D transforms work and are having trouble understanding the description because of that, or that you interpret the source as a target the wind is blowing *towards* not a source

**Comment by yankscally** [comment]
yeah i keep saying source because its a better term for me to understand. its the "source" of the wind direction. even if it blows the opposite way. its a node that controls the wind.

3D transforms complicates what the target node needs to do if its only about rotation. transform implies transforming

the docs don't need to mention transforms just needs to say something like: "wind will blow in the opposite direction to the target's _rotation_. The wind direction is the opposite to the rotation."

**Comment by AThousandShips** [comment]
> 3D transforms complicates what the target node needs to do if its only about rotation. transform implies transforming

Rotation is part of transforming, I think you are thinking about *translation* which is *also* part of transforming, but rotation *is* part of the transform

> the docs don't need to mention transforms just needs to say something like: "wind will blow in the opposite direction to the target's rotation. The wind direction is the opposite to the rotation."

But that would be wrong, it's not the rotation, it's the z-axis of the transform.
> The direction is opposite to the z-axis of the Node3D's local transform

I think you should try start thinking about it as a source not a target, that should help

**Comment by yankscally** [comment]
I think what was happening - I was changing the rotation as the project was running, expecting it to change visually, but the area has to re-enter the subject of wind, as you mentioned in the bug report, which isn't stated in the docs - so it looked like a bug to me as I was expecting to be able to actually change the wind value at runtime. 

Now that I can use this, it only seems like the X and Y rotation really make a difference - also similar to how directional light works. I think _that_ is what should be what is explained, as well as the Area3D needing to re-enter the softbody to give it a new wind direction. This kind of makes the wind feature only useful if it intends to be static, which most of the time it is not.

**Comment by MartinDelille** [comment]
Would it explain the issue I encounter here ? https://github.com/godotengine/godot-docs-user-notes/discussions/155#discussioncomment-12742402

**Comment by yankscally** [comment]
@MartinDelille 
Yes, the body must leave the area and re enter to change the wind direction.

Also. The wind is from a direction like the directional light

**Comment by LucidTaZ** [comment]
I don't recognize this issue from my own experience. I'm creating a sailing simulator that relies on SoftBody3D with wind in an Area3D for visuals. The Area3D travels with the vessel (it's a child node), just like the SoftBody3D (sail model). This means that the SoftBody3D never leaves or re-enters the Area3D. Still, the wind works as expected and I can change the direction without problems.

Perhaps the fact that the Area3D is a descendant of the vessel is what makes the difference? It means that the global transform of the Area3D is updated every frame. Caveat: I need to reset the rotation of the Area3D to zero on every frame, otherwise the wind is off.

I've recorded a video to demonstrate this. I'm changing the wind direction with a slider in the UI of the game. I've added a visualization of the Area3D's wind origin XYZ axes to show that the origin only rotates around its Y axis. I'm not using any local translation on it.
https://github.com/user-attachments/assets/e028fdab-6273-4bb6-aa56-b03014c05902

**Comment by yankscally** [comment]
@LucidTaZ oh, this was reported using the original Godot Physics. 

Are you using Jolt? It may have been fixed there or somewhere else since. 

Really nice game btw!

**Comment by yankscally** [comment]
@LucidTaZ 

That's really odd. I remember spending a few hours on this issue. I should get a video if I can, or else I will just close the issue.

**Comment by Blubtastic** [comment]
I'm having similar issues - changing the rotation of the Node3D used as the Source Path works when changing it in the editor, but not through code (although it updates in the editor in the "Remote" tree while running the game). 

I might have misunderstood something, but I've tried almost everything and nothing makes it possible to update the direction in runtime. My setup is basically the same as @LucidTaZ.

**Comment by LucidTaZ** [comment]
Strange. I'm changing the wind origin in `_physics_process` and checked that it also works in `_process`.

Since my previous comment on this issue, I've moved the Area3D to travel with the camera, not with the SoftBody3D. It still works fine, perhaps still because it updates every frame.

It would help to get a minimal reproduction project, see if I can reproduce the issue on my machine.

**Comment by Blubtastic** [comment]
That would be great, I'm not that familiar with how to best share the setup, but I'd assume just cloning/downloading the github project and opening it in Godot will work: https://github.com/Blubtastic/sailing_simulator

To test wind directions, change `wind_direction` in the `globals.gd` script. The `WindForSails.tsnc` scene is where the area3D wind stuff is. 
 

I'm using Godot physics

- Godot v4.4.1.stable.official
- MacOS 15.5
_Note: I'm developing in Compatibility mode, so remember to use it. I don't think that's why I'm having issues._

Edit: here's a minimal reproduction project: https://github.com/Blubtastic/AlignParentToWindBug

**Comment by LucidTaZ** [comment]
I think I found a workaround for this problem, using @Blubtastic's project (which looks really nice by the way) and finding out what's different from mine.

💡 Assigning a value to the Area3D's `wind_force_magnitude` variable on every physics frame, or in any case after rotating the source node, fixes the issue. The wind will now properly update and influence the SoftBody3D's forces. Even using `wind_force_magnitude = wind_force_magnitude` will do the trick.

@Blubtastic furthermore I figured out that your invocation (only for the wind origin, not the other parts of your game) should then become:
```gdscript
parent.look_at(parent.global_position + wind_direction)
# instead of:
parent.look_at(parent.global_position - wind_direction)
```
but that's something you can experiment with yourself.

**Comment by Blubtastic** [comment]
That worked, thanks a lot for your help!
Maybe this could also be of help to @yankscally's problem?

Your bottom code suggestion sounds right as well @LucidTaZ. And your project looks really professional, hope you don't mind me taking some inspiration from it.

**Comment by yankscally** [comment]
> Maybe this could also be of help to [@yankscally](https://github.com/yankscally)'s problem?

I think the problem is I expected it to update automatically with the Area3D nodes rotation, so it felt like a possible bug. and I think this should be documented to avoid confusion:

> Assigning a value to the Area3D's `wind_force_magnitude` variable on every physics frame, or in any case after rotating the source node, fixes the issue.
